src/lib/auth.ts

```diff
@@ -5,55 +5,61 @@ import { NextAuthOptions } from "next-auth";
 import GoogleProvider from "next-auth/providers/google";
 
 export const authOptions: NextAuthOptions = {
+  pages: {
+    signIn: "/",
+    signOut: "/",
+    error: "/",
+  },
   callbacks: {
     async signIn({ user }) {
       if (!user.email) throw new Error("user must have an email");
-      await prisma.user.upsert({
-        where: {
-          id: user.id,
-        },
-        update: {
-          name: user.name,
-        },
-        create: {
-          name: user.name,
-          email: user.email,
-          id: user.id,
-        },
-      });
+
       return true;
     },
 
     async redirect({ url }) {
       return url;
     },
+
     async jwt({ token, account, user }) {
       if (account) {
         token.accessToken = account.access_token;
         token.sub = user.id;
       }
+
       return token;
     },
+
     async session({ session, token }) {
       if (token) {
         session.user.id = token.sub ?? "";
-        // Put the default planner id of the current user into the session context
-        session.user.defaultPlannerId = (
-          await prisma.user.findFirst({
-            where: {
-              id: session.user.id,
-            },
-            select: {
-              defaultPlannerId: true,
-            },
-          })
-        )?.defaultPlannerId;
+
+        // QUESTION: Why is default planner id being set here? Should be in user
```

Comment:
It's helpful to have the default planner id in the session object to avoid having to refetch it again later on

Reply:
Doesn't the default planner id change if you were to delete it? Mutable user data shouldn't be stored in the session, if you want it to always be available you can use Zustand store or react query if it is server state and set a stale time of infinity. 

Maybe I misunderstand the point of this value.

Reply:
No you're right, this is probably a cause of some subtle errors. I think it would be best to just use the `useUserDefaultPlannerId` hook to access it as it is technically server state